api/views.py

```diff
@@ -28,6 +30,49 @@ def get(self, request, pk):
         return Response(serializer.data)
 
 
+class AnswerViewSet(viewsets.ModelViewSet):
+    permission_classes = [IsAuthenticated]
+    serializer_class = AnswerSerializer
+
+    def __sanitize_answer(self, answer):
+        """Strips whitespace and converts to uppercase."""
+        return re.sub(r"\s", "", answer).upper()
+
+    def get_queryset(self):
+        puzzle_id = self.kwargs["puzzle_id"]
+        return Answer.objects.filter(puzzle__id=puzzle_id)
+
+    def create(self, request, **kwargs):
+        puzzle = None
+        with transaction.atomic():
+            hunt = get_object_or_404(Hunt, pk=self.kwargs["hunt_id"])
+            puzzle = get_object_or_404(Puzzle, pk=self.kwargs["puzzle_id"])
+            serializer = self.get_serializer(data=request.data)
+            serializer.is_valid(raise_exception=True)
+            text = self.__sanitize_answer(serializer.validated_data["text"])
+            answer, created = Answer.objects.get_or_create(text=text, puzzle=puzzle)
```

Comment:
I think the serializer has a couple methods like `serializer.create()` or `serializer.save()` which we could also use though in this case this seems fine as well.